opm/simulators/wells/MSWellHelpers.hpp

```diff
@@ -62,8 +68,18 @@ namespace mswellhelpers
                 }
             }
         }
-
         return y;
+    }
+#endif
+
+    // obtain y = D^-1 * x with a direct solver
+    template <typename MatrixType, typename VectorType>
+    VectorType
+    invDXDirect(const MatrixType& D, VectorType x)
```

Comment:
I do not think this function is used anymore. It is good to remove it completely to avoid confusion.  

And also, since this function is not used anymore, the checking of the `HAVE_UMFPACK` is also lost.  Please also keep it when removing this function.